### ISSUE TITLE
Price Lists with Min_quantity and Max_quantity Constraints Not Functioning Properly

### ISSUE THREAD

**Issue by Rask777** [medusajs/medusa#3584]
### Describe the bug

Scenario 1:  Min_quantity in active price list not reflected in cart price
When the min_quantity value is set for a product in the active price list, the cart does not update the price accordingly. This leads to incorrect price calculations in the cart when the product quantity is below the specified min_quantity.

Scenario 2: Max_quantity in the active price list affects the price for all quantities
When the max_quantity value is set for a product in the active price list, the cart updates the price accordingly for all quantities, even if the quantity in the cart is less than the specified max_quantity. This leads to incorrect price calculations in the cart for product quantities, not within the specified range.

### System information

Medusa version (including plugins): v1.7.15 & v1.8.0-rc.0
nextjs-starter-medusa: V 1.0.3
Database: Postgres 15
Operating system: MacOS

### Steps to reproduce the behavior

Scenario 1:
- Create a new product or select an existing product.
- Add the product to the active price list and set a min_quantity value.
- Add the product to the cart with a quantity less than the min_quantity value.
- Observe the price in the cart.

Scenario 2:
- Create a new product or select an existing product.
- Add the product to the active price list and set a max_quantity value.
- Add the product to the cart with a quantity less than the max_quantity value.
- Observe the price in the cart.

### Expected behavior

Scenario 1:
The cart should calculate the price considering the min_quantity constraint set in the active price list. If the product quantity is less than the min_quantity, the cart should display the regular price without any discount. If the product quantity is equal to or greater than the min_quantity, the cart should apply the appropriate discount as specified in the price list.

Scenario 2:
The cart should calculate the price based on the max_quantity set in the active price list, applying the special price only when the product quantity is within the specified range.

### Code snippets

<details>
<summary>response from localhost:9000/store/carts/{cart_i}/line-items/{line_item}</summary>

```
  `{
   "cart":{
      "object":"cart",
      "id":"cart_01GWDHWQA8Q2W0VMQ5JRH9N2GX",
      "created_at":"2023-03-25T23:33:07.522Z",
      "updated_at":"2023-03-25T23:33:07.522Z",
      "deleted_at":null,
      "email":null,
      "billing_address_id":null,
      "shipping_address_id":null,
      "region_id":"reg_01GWBD5SFT92DNV2CPEYB4XM9S",
      "customer_id":null,
      "payment_id":null,
      "type":"default",
      "completed_at":null,
      "payment_authorized_at":null,
      "idempotency_key":null,
      "context":{
         "ip":"::1",
         "user_agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"
      },
      "metadata":null,
      "sales_channel_id":"sc_01GWAVRF5887EXDH9498DVG0K1",
      "gift_cards":[
         
      ],
      "region":{
         "id":"reg_01GWBD5SFT92DNV2CPEYB4XM9S",
         "created_at":"2023-03-25T03:32:12.881Z",
         "updated_at":"2023-03-25T03:32:12.881Z",
         "deleted_at":null,
         "name":"EU",
         "currency_code":"eur",
         "tax_rate":0,
         "tax_code":null,
         "gift_cards_taxable":true,
         "automatic_taxes":true,
         "tax_provider_id":null,
         "metadata":null,
         "countries":[
            {
               "id":109,
               "iso_2":"it",
               "iso_3":"ita",
               "num_code":380,
               "name":"ITALY",
               "display_name":"Italy",
               "region_id":"reg_01GWBD5SFT92DNV2CPEYB4XM9S"
            },
            {
               "id":209,
               "iso_2":"es",
               "iso_3":"esp",
               "num_code":724,
               "name":"SPAIN",
               "display_name":"Spain",
               "region_id":"reg_01GWBD5SFT92DNV2CPEYB4XM9S"
            },
            {
               "id":75,
               "iso_2":"fr",
               "iso_3":"fra",
               "num_code":250,
               "name":"FRANCE",
               "display_name":"France",
               "region_id":"reg_01GWBD5SFT92DNV2CPEYB4XM9S"
            },
            {
               "id":60,
               "iso_2":"dk",
               "iso_3":"dnk",
               "num_code":208,
               "name":"DENMARK",
               "display_name":"Denmark",
               "region_id":"reg_01GWBD5SFT92DNV2CPEYB4XM9S"
            },
            {
               "id":82,
               "iso_2":"de",
               "iso_3":"deu",
               "num_code":276,
               "name":"GERMANY",
               "display_name":"Germany",
               "region_id":"reg_01GWBD5SFT92DNV2CPEYB4XM9S"
            },
            {
               "id":235,
               "iso_2":"gb",
               "iso_3":"gbr",
               "num_code":826,
               "name":"UNITED KINGDOM",
               "display_name":"United Kingdom",
               "region_id":"reg_01GWBD5SFT92DNV2CPEYB4XM9S"
            },
            {
               "id":215,
               "iso_2":"se",
               "iso_3":"swe",
               "num_code":752,
               "name":"SWEDEN",
               "display_name":"Sweden",
               "region_id":"reg_01GWBD5SFT92DNV2CPEYB4XM9S"
            }
         ],
         "payment_providers":[
            {
               "id":"manual",
               "is_installed":false
            }
         ],
         "tax_rates":[
            
         ],
         "fulfillment_providers":[
            {
               "id":"manual",
               "is_installed":false
            }
         ]
      },
      "items":[
         {
            "id":"item_01GWDYDBXTMGRA2NDX1RW78VJS",
            "created_at":"2023-03-26T03:11:55.811Z",
            "updated_at":"2023-03-26T03:15:27.628Z",
            "cart_id":"cart_01GWDHWQA8Q2W0VMQ5JRH9N2GX",
            "order_id":null,
            "swap_id":null,
            "claim_order_id":null,
            "original_item_id":null,
            "order_edit_id":null,
            "title":"Medusa Coffee Mug",
            "description":"One Size",
            "thumbnail":"https://medusa-public-images.s3.eu-west-1.amazonaws.com/coffee-mug.png",
            "is_return":false,
            "is_giftcard":false,
            "should_merge":true,
            "allow_discounts":true,
            "has_shipping":false,
            "unit_price":1000,
            "variant_id":"variant_01GWBD5SPP3VZCGYHQJNNRC9EN",
            "quantity":7,
            "fulfilled_quantity":null,
            "returned_quantity":null,
            "shipped_quantity":null,
            "metadata":{
               
            },
            "variant":{
               "id":"variant_01GWBD5SPP3VZCGYHQJNNRC9EN",
               "created_at":"2023-03-25T03:32:12.881Z",
               "updated_at":"2023-03-25T03:32:12.881Z",
               "deleted_at":null,
               "title":"One Size",
               "product_id":"prod_01GWBD5SPKGP726PYE272G3RCN",
               "sku":null,
               "barcode":null,
               "ean":null,
               "upc":null,
               "variant_rank":0,
               "inventory_quantity":100,
               "allow_backorder":false,
               "manage_inventory":true,
               "hs_code":null,
               "origin_country":null,
               "mid_code":null,
               "material":null,
               "weight":null,
               "length":null,
               "height":null,
               "width":null,
               "metadata":null,
               "product":{
                  "id":"prod_01GWBD5SPKGP726PYE272G3RCN",
                  "created_at":"2023-03-25T03:32:12.881Z",
                  "updated_at":"2023-03-25T03:45:01.962Z",
                  "deleted_at":null,
                  "title":"Medusa Coffee Mug",
                  "subtitle":null,
                  "description":"Every programmer's best friend.",
                  "handle":"coffee-mug",
                  "is_giftcard":false,
                  "status":"published",
                  "thumbnail":"https://medusa-public-images.s3.eu-west-1.amazonaws.com/coffee-mug.png",
                  "profile_id":"sp_01GWAVRF4XQ8RB26BM3C3YZHN2",
                  "weight":400,
                  "length":null,
                  "height":null,
                  "width":null,
                  "hs_code":null,
                  "origin_country":null,
                  "mid_code":null,
                  "material":null,
                  "collection_id":"pcol_01GWBDWH8K9T765TNMM1N100RJ",
                  "type_id":null,
                  "discountable":true,
                  "external_id":null,
                  "metadata":null
               }
            },
            "adjustments":[
               
            ],
            "tax_lines":[
               {
                  "rate":0,
                  "name":"default",
                  "code":"default",
                  "item_id":"item_01GWDYDBXTMGRA2NDX1RW78VJS"
               }
            ],
            "subtotal":7000,
            "discount_total":0,
            "total":7000,
            "original_total":7000,
            "original_tax_total":0,
            "tax_total":0
         }
      ],
      "payment":null,
      "shipping_address":null,
      "billing_address":null,
      "shipping_methods":[
         
      ],
      "payment_sessions":[
         
      ],
      "discounts":[
         
      ],
      "sales_channel":{
         "id":"sc_01GWAVRF5887EXDH9498DVG0K1",
         "created_at":"2023-03-24T22:27:50.531Z",
         "updated_at":"2023-03-24T22:27:50.531Z",
         "deleted_at":null,
         "name":"Default Sales Channel",
         "description":"Created by Medusa",
         "is_disabled":false
      },
      "subtotal":7000,
      "discount_total":0,
      "item_tax_total":0,
      "shipping_total":0,
      "shipping_tax_total":0,
      "gift_card_total":0,
      "gift_card_tax_total":0,
      "tax_total":0,
      "total":7000
   }
}`
```

</details>

### Additional context

Several individuals in the Medusa Discord have brought up this issue, but I have been unable to locate an open issue addressing it.

**Comment by monkeyMax-SVG** [comment]
I am getting same issue.

**Comment by lucassilver07** [comment]
I am getting exactly same issue.
Also, in my admin side, i have no min_quantity, max_quantity interface . so I added it myself. 
All metadata seems working but once I set min & max quantity in price list, the new price is not applied on product detail page and other pages.

**Comment by lucassilver07** [comment]
I am getting same issue.
I just installed new medusa app using create-medusa-app

1. First of all, there is missing part (interface to set min_quantity and max_quantity configurations)
![Screenshot 2023-04-20 at 19 18 57](https://user-images.githubusercontent.com/114839922/233350781-0ea8deaf-c090-4476-8ca6-2a2ad19aca50.png)

2. I added min_quantity and max_quantity interface myself.
Here, I have found some interesting things.
max_quantity never works.
min_quantity works only if its value is 0 or 1.
- If I set it as zero (0), it shows the applied price in products/[handle] page and /cart page
- If I set it as one (1), it shows only in /cart page but not in products/[handle] page.
